monkey/common/credentials/credentials.py

```diff
@@ -139,6 +162,41 @@ def from_json(credentials: str) -> Credentials:
         except MarshmallowError as err:
             raise InvalidCredentialsError(str(err))
 
+    @staticmethod
+    def from_json_array(credentials_array_json: str) -> Sequence[Credentials]:
```

Comment:
This seems redundant with the `from_json`. Whoever sends a single credential object can also put that into an array.

Reply:
Yep but if you need to do it in a multiple places then it becomes code duplication.

Reply:
Yeah, it's for convenience and to reduce code duplication. If it turns out we don't use it, we can remove it.

Reply:
It's not a responsibility of the credentials dataclass to solve code duplication in the callers, it's to provide a good interface. I also don't understand how redundancy is better than duplication.

Reply:
The logic is out of place here, IMO. This is a credentials object definition, it's not about parsing json lists.